client/service/src/task_manager/tests.rs

```diff
@@ -37,6 +37,7 @@ impl DropTester {
 	}
 
 	fn wait_on_drop(&self) {
+		#[allow(clippy::while_immutable_condition)]
```

Comment:
This is not true.

Reply:
This is a bug in clippy, because if you read the code, you will see that this is not an immutable while condition.